src/main/java/org/jabref/model/entry/BibEntryTypeBuilder.java

```diff
@@ -20,17 +23,25 @@
 public class BibEntryTypeBuilder {
 
     private EntryType type = StandardEntryType.Misc;
-    private SequencedSet<BibField> fields = new LinkedHashSet<>();
+
     private SequencedSet<OrFields> requiredFields = new LinkedHashSet<>();
+    private SequencedSet<BibField> optionalFields = new LinkedHashSet<>();
+
+    private Set<Field> seenFields = new HashSet<>();
```

Comment:
What is seen fields now?

Reply:
We are in the context of the **Builder**. The builder should raise an IllegalArgumentException of arguments are wrong. Therefore, the builder needs to know what it has seen.